src/Runner.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -634,28 +634,34 @@ private async Task<GitHubRunnerRegisterToken> GetJITRunnerTokenAsync(string gith
                             var jsonResponse = await response.Content.ReadAsStringAsync();
                             return StringUtil.ConvertFromJson<GitHubRunnerRegisterToken>(jsonResponse);
                         }
-                        else if(response.StatusCode == System.Net.HttpStatusCode.NotFound)
-                        {
-                            // It doesn't make sense to retry in this case, so just stop
-                            break;
-                        }
                         else
                         {
                             _term.WriteError($"Http response code: {response.StatusCode} from 'POST {githubApiUrl}'");
                             var errorResponse = await response.Content.ReadAsStringAsync();
                             _term.WriteError(errorResponse);
-                            response.EnsureSuccessStatusCode();
+
```

Comment:
Can we just try storing the response code and using that to determine if we should throw or not.

```
                    var responseStatus = HttpStatusCode.OK;
                    try
                    {
                        var response = await httpClient.PostAsync(githubApiUrl, new StringContent(string.Empty));
                        responseStatus = response.StatusCode;
                        if (response.IsSuccessStatusCode)
                        {
                            Trace.Info($"Http response code: {response.StatusCode} from 'POST {githubApiUrl}'");
                            var jsonResponse = await response.Content.ReadAsStringAsync();
                            return StringUtil.ConvertFromJson<GitHubRunnerRegisterToken>(jsonResponse);
                        }
                        else
                        {
                            _term.WriteError($"Http response code: {response.StatusCode} from 'POST {githubApiUrl}'");
                            var errorResponse = await response.Content.ReadAsStringAsync();
                            _term.WriteError(errorResponse);
                            response.EnsureSuccessStatusCode();
                        }
                    }
                    catch(Exception ex) when (retryCount < 2 && responseStatus != HttpStatusCode.NotFound)
                    {
                        retryCount++;
                        Trace.Error($"Failed to get JIT runner token -- Atempt: {retryCount}");
                        Trace.Error(ex);
                        Trace.Info("Retrying in 5 seconds");
                    }
```

Reply:
We may want to add others to this not retry list, and looking at the response code is the easiest way to do that

Reply:
Sure, makes sense.

Reply:
I just won't trace delay time there, because it's randomly determined outside of catch.